operator/api/redpanda/v1alpha3/redpanda_conversion.go

```diff
@@ -0,0 +1,954 @@
+		}
+
+	case EmbeddedNodePoolSpec:
+		nodeConfig := mapValues(values.Config.Node, func(v any) ValueSource {
+			out, err := yaml.Marshal(v)
+			if err != nil {
+				panic(err)
+			}
+
+			return ValueSource{
+				Value:           (*YAMLRepresentation)(ptr.To(string(out))),
+				ConfigMapKeyRef: nil,
+				SecretKeyRef:    nil,
+				UseRawValue:     false,
+			}
+		})
+
+		rpkConfig := mapValues(values.Config.RPK, func(v any) ValueSource {
+			out, err := yaml.Marshal(v)
+			if err != nil {
+				panic(err)
+			}
+
+			return ValueSource{
+				Value:           (*YAMLRepresentation)(ptr.To(string(out))),
+				ConfigMapKeyRef: nil,
+				SecretKeyRef:    nil,
+				UseRawValue:     false,
+			}
+		})
+
+		image := ""
+		if values.Image.Tag != "" {
+			image = fmt.Sprintf("%s:%s", values.Image.Repository, values.Image.Tag)
+		} else if repo := get[*string](src.Spec.ClusterSpec, "Image", "Repository"); repo != nil {
+			image = *repo
+		}
+
+		return EmbeddedNodePoolSpec{
+			Replicas: get[*int32](src.Spec.ClusterSpec, "Statefulset", "Replicas"),
+			BrokerTemplate: BrokerTemplate{
+				Image:                     image,
+				RPKConfig:                 rpkConfig,
+				NodeConfig:                nodeConfig,
+				Resources:                 values.Resources.GetResourceRequirements(),
+				ValidateFilesystem:        values.Statefulset.InitContainers.FSValidator.Enabled,
+				SetDataDirectoryOwnership: values.Statefulset.InitContainers.SetDataDirOwnership.Enabled,
+				PodTemplate:               convertFrom[*PodTemplate](src, values).(*PodTemplate),
+			},
+		}
+
+	case *PodTemplate:
+		pt := get[*redpandav1alpha2.PodTemplate](src.Spec.ClusterSpec, "Statefulset", "PodTemplate")
+
+		if pt == nil {
+			return (*PodTemplate)(nil)
+		}
+
+		return &PodTemplate{
+			ObjectMeta: ObjectMeta{
+				Labels:      pt.Labels,
+				Annotations: pt.Annotations,
+			},
+			Spec: (*applycorev1.PodSpecApplyConfiguration)(pt.Spec),
+		}
+
+	case RedpandaStatus:
+		return RedpandaStatus{
+			Conditions:    src.Status.Conditions,
+			ConfigVersion: src.Status.ConfigVersion,
+			LicenseStatus: (*LicenseStatus)(src.Status.LicenseStatus),
+			NodePools: doMap(src.Status.NodePools, func(src redpandav1alpha2.NodePoolStatus) NodePoolStatus {
+				return NodePoolStatus{
+					Name:              src.Name,
+					Replicas:          src.Replicas,
+					DesiredReplicas:   src.DesiredReplicas,
+					OutOfDateReplicas: src.OutOfDateReplicas,
+					UpToDateReplicas:  src.UpToDateReplicas,
+					CondemnedReplicas: src.CondemnedReplicas,
+					ReadyReplicas:     src.ReadyReplicas,
+					RunningReplicas:   src.RunningReplicas,
+				}
+			}),
+		}
+
+	default:
+		panic(fmt.Sprintf("unhandled type in convFrom: %T", t))
+	}
+}
+
+func convertTo[T any](src *Redpanda) any {
+	t := reflect.New(reflect.TypeFor[T]()).Elem().Interface()
+
+	switch t.(type) {
+	case redpandav1alpha2.Redpanda:
+		return redpandav1alpha2.Redpanda{
+			TypeMeta:   metav1.TypeMeta{APIVersion: "", Kind: ""},
```

Comment:
see above about TypeMeta, drop or specify, the empty strings don't make sense to me.

Reply:
Addressed above and added a comment:

```
			// Specified to sate exhaustruct. This is otherwise the zero value,
			// which is what is generally encountered on a K8s Object not fresh
			// from the API or in a gotohelm chart.

```